react-stopwatch/my-app/src/Stopwatch.tsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react';
+import './App.css';
+import './Stopwatch.css';
+import { FaPlay, FaPause } from 'react-icons/fa';
+
+export function Stopwatch() {
+  const [time, setTime] = useState(0);
+  const [intervalFunction, setIntervalFunction] = useState<NodeJS.Timeout>();
+  const running = intervalFunction !== undefined;
+
+  return (
+    <>
+      <div className="watch-edge">
+        <h2
+          className="timer"
+          onClick={() => {
+            if (!running) setTime(0);
+          }}>
+          {time}
+        </h2>
+      </div>
+      {running ? (
+        <FaPause
+          onClick={() => {
```

Comment:
From a readability standpoint, it's nice to have only very short arrow functions in your JSX. So if the function has more than a line (or two, in extreme cases), it is preferable to put it into a nested function inside the component.

Reply:
Fair enough, I did go a bit overboard on what I ought to keep inline.